packages/flutter/lib/src/rendering/sliver_list.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:meta/meta.dart';
+
+import 'box.dart';
+import 'sliver.dart';
+import 'sliver_multi_box_adaptor.dart';
+
+class RenderSliverList extends RenderSliverMultiBoxAdaptor {
+  RenderSliverList({
+    @required RenderSliverBoxChildManager childManager,
+    double itemExtent,
+  }) : _itemExtent = itemExtent, super(childManager: childManager);
+
+  /// The main-axis extent of each item in the list.
+  double get itemExtent => _itemExtent;
+  double _itemExtent;
+  set itemExtent (double newValue) {
+    assert(newValue != null);
+    if (_itemExtent == newValue)
+      return;
+    _itemExtent = newValue;
+    markNeedsLayout();
+  }
+
+  double _indexToScrollOffset(int index) => _itemExtent * index;
+
+  @override
+  void performLayout() {
+    assert(debugAssertNotCurrentlyAllowingChildAdditions());
+    final double scrollOffset = constraints.scrollOffset;
+    assert(scrollOffset >= 0.0);
+    final double remainingPaintExtent = constraints.remainingPaintExtent;
+    assert(remainingPaintExtent >= 0.0);
+    final double targetEndScrollOffset = scrollOffset + remainingPaintExtent;
+
+    BoxConstraints childConstraints = constraints.asBoxConstraints(
+      minExtent: itemExtent,
+      maxExtent: itemExtent,
+    );
+
+    final int firstIndex = math.max(0, scrollOffset ~/ _itemExtent);
+    final int targetLastIndex = math.max(0, (targetEndScrollOffset / itemExtent).ceil());
+
+    if (firstChild != null) {
+      final int oldFirstIndex = indexOf(firstChild);
+      final int oldLastIndex = indexOf(lastChild);
+      final int leadingGarbage = math.max(0, firstIndex - oldFirstIndex);
+      final int trailingGarbage = math.max(0, oldLastIndex - targetLastIndex);
+      if (leadingGarbage + trailingGarbage > 0)
+        collectGarbage(leadingGarbage, trailingGarbage);
+    }
+
+    if (firstChild == null) {
+      if (!addInitialChild(index: firstIndex, scrollOffset: _indexToScrollOffset(firstIndex))) {
+        // There are no children.
+        geometry = new SliverGeometry(
```

Comment:
i think these are the default values for those arguments, you can just omit them